jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,43 +149,173 @@ public static boolean isCRIUSupportEnabled() {
+		return this;
+	}
+
+	/**
+	 * Sets the verbosity of log output. Corresponds to criu_set_log_level.
+	 * Default: 2 (errors + warnings)
+	 * 
+	 * @param logLevel verbosity from 1 to 4 inclusive
+	 * @return this
+	 * @throws IllegalArgumentException if logLevel is not valid.
+	 */
+	public CRIUSupport setLogLevel(int logLevel) {
+		if (logLevel > 0 && logLevel <= 4) {
+			this.logLevel = logLevel;
+		} else {
+			throw new IllegalArgumentException("Log level must be 1 to 4 inclusive");
+		}
+		return this;
+	}
+
+	/**
+	 * Write log output to logFile. Corresponds to criu_set_log_file.
+	 * Default: criu.log
+	 * 
+	 * @param logFile name of the file to write log output to. The path to the file can be set with {@link #setWorkDir(Path)}.
+	 * @return this
+	 * @throws IllegalArgumentException if logFile is null or a path.
+	 */
+	public CRIUSupport setLogFile(String logFile) {
+		if (logFile != null && !logFile.contains(File.separator)) {
+			this.logFile = logFile;
+		} else {
+			throw new IllegalArgumentException("Log file must not be null and not be a path");
+		}
+		return this;
+	}
+
+	/**
+	 * Controls whether to dump file locks. Corresponds to criu_set_file_locks.
+	 * Default: false
+	 * 
+	 * @param fileLocks
+	 * @return this
+	 */
+	public CRIUSupport setFileLocks(boolean fileLocks) {
+		this.fileLocks = fileLocks;
+		return this;
+	}
+
+	/**
+	 * Sets the directory where non-image files are stored (e.g. logs). Corresponds to criu_set_work_dir_fd.
+	 * Default: same as path set by {@link #setImagesDir(Path)}.
+	 * 
+	 * @param workDir the directory as a java.nio.file.Path
+	 * @return this
+	 * @throws NullPointerException if workDir is null.
+	 * @throws SecurityException if no permission to access workDir.
+	 * @throws IllegalArgumentException if workDir is not a valid directory.
+	 */
+	public CRIUSupport setWorkDir(Path workDir) {
+		Objects.requireNonNull(workDir, "Path cannot be null");
+		if (!Files.isDirectory(workDir)) {
+			throw new IllegalArgumentException("workDir is not a valid directory");
+		}
+		String dir = workDir.toAbsolutePath().toString();
+
+		SecurityManager manager = System.getSecurityManager();
+		if (manager != null) {
+			manager.checkWrite(dir);
+		}
+
+		this.workDir = dir;
+		return this;
+	}
+
 	/**
 	 * Checkpoint the JVM. A security manager check is done for org.eclipse.openj9.criu.CRIUDumpPermission
 	 * as well as a java.io.FilePermission check to see if checkPointDir is writable.
 	 * 
-	 * This operation will re-initialize the CRIU options
+	 * This operation will use the CRIU options set by the options setters.
 	 * 
 	 * All errors will be stored in the throwable field of CRIUResult.
 	 * 
-	 * @param checkPointDir the directory in which the checkpoint data will be stored
-	 * 
 	 * @return return CRIUResult
 	 */
-	public static CRIUResult checkPointJVM(Path checkPointDir) {
-		CRIUResult ret = new CRIUResult(CRIUResultType.UNSUPPORTED_OPERATION, null);
+	public CRIUResult checkPointJVM() {
```

Comment:
The name should start with `checkpoint` (the javadoc suggests 'checkpoint' is one word).